### PR TITLE
Fix warning on spaces log (#9317)

### PR DESCRIPTION
**Pull request by olivierchatry**

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -502,6 +502,11 @@ function WebGLRenderer( parameters ) {
 
 	this.dispose = function () {
 
+		transparentObjects = [];
+		transparentObjectsLastIndex = -1;
+		opaqueObjects = [];
+		opaqueObjectsLastIndex = -1;
+
 		_canvas.removeEventListener( 'webglcontextlost', onContextLost, false );
```
Comment (takahirox) [inline src/renderers/WebGLRenderer.js]: What's this for?

**Comment by olivierchatry** [comment]
This is super weird, old commit that have already been integrated I think (
was a memory leak when disposing / recreating )

**Comment by Mugen87** [comment]
Do you want to solve the following issue with your PR? #11138

If so, i think it's better to fix this on Chrome side.

**Comment by Mugen87** [comment]
I've realized we have two issues for the same thing. I'll close #9317 so #11138 is the leading one.

**Comment by olivierchatry** [comment]
Hum, you mean that chrome should do what the PR does ? I did it because it
was bothersome, and can be easily remove if chrome fix it.

**Comment by Mugen87** [comment]
There is already a filed bug at the chromium project. IMO, it makes more sense to request an implementation of this specific issue.

https://bugs.chromium.org/p/chromium/issues/detail?id=711520

**Comment by olivierchatry** [comment]
Well, it took me 3 seconds to make this fix, we are already discussing it
for more than 5 minutes and the chrome bug have been filled in april, with
people arguing if it is a threejs issue, or whatever.

This is this code that needs to do the same thing as I do :

https://cs.chromium.org/chromium/src/third_party/WebKit/Source/modules/webgl/WebGLRenderingContextBase.cpp?sq=package:chromium&l=3248

**Comment by olivierchatry** [comment]
From this one :

https://cs.chromium.org/chromium/src/third_party/WebKit/Source/modules/webgl/GLStringQuery.h?sq=package:chromium&l=17

Anyway the return value is driver dependant. So if a manufacturer decide to
fill with '*' when there is nothing to be reported, I think it should be
the 3d engine, not the webgl implementation, that fix that.

**Comment by olivierchatry** [comment]
Also, *@takahirox I realized I never did a pull request for the other bug
... *

**Comment by Mugen87** [comment]
@kenrussell Any chances of a soon fix for this issue?

https://bugs.chromium.org/p/chromium/issues/detail?id=711520

**Comment by olivierchatry** [comment]
IMHO, I think it should be the 3d engine, not the webgl low level
implementation that fix that. But it should also be a 5 minutes fix  ( at
least the code, not the whole regression testing ) in blink code.

**Comment by Mugen87** [comment]
@olivierchatry I just want to add that i generally prefer a fix in Chrome. But if the Chromium guys won't respond, i vote for a fix in the library. In any event, thanks for pushing this issue forward 👍

**Comment by Mugen87** [comment]
Okay, i vote for a change in the library. It does not look like we get a fix for Chrome in the near future. But we only should merge this PR without the following commit: 513fa23086d1ced9f51b62476c4b5d9377807ea1

**Comment by olivierchatry** [comment]
Well not sure what is going on with this one, I think it was already
integrated.

https://github.com/mrdoob/three.js/issues/8987
https://github.com/mrdoob/three.js/pull/9011

File: src/renderers/webgl/WebGLProgram.js [hunk 2]
```diff
@@ -540,9 +540,9 @@ function WebGLProgram( renderer, extensions, code, material, shader, parameters
 
 	gl.linkProgram( program );
 
-	var programLog = gl.getProgramInfoLog( program );
-	var vertexLog = gl.getShaderInfoLog( glVertexShader );
-	var fragmentLog = gl.getShaderInfoLog( glFragmentShader );
+	var programLog = gl.getProgramInfoLog( program ).replace(/\0/g, '').trim();
```
Comment (haroldiedema) [inline src/renderers/webgl/WebGLProgram.js]: This isn't going to fix this issue for all cases.

As the developers at chromium stated, the information here is driver implementation specific. In my case, this returns 4 blank lines on some occasions.

Also maybe it's not a good idea to use regex in this part of the code for performance reasons.

**Review by haroldiedema — CHANGES_REQUESTED** [review]

**Comment by olivierchatry** [comment]
From my test, it should solve your issue ( trim take cares of the empty
lines as well ). I tested it in nodejs, but I might be wrong. For the
performance bit, I would argue that it will not really impact anything, but
I'm sure a profile will give way better number.

**Comment by Mugen87** [comment]
see #12950

**Pull request closed without merging**